ably.d.ts

```diff
@@ -268,106 +543,249 @@ declare namespace Types {
+		 */
 		reason?: ErrorInfo;
+		/**
+		 * Duration upon which the client will retry a connection where applicable, as milliseconds.
+		 */
 		retryIn?: number;
 	}
 
+	/**
+	 * A type encapsulating attributes of a device registered for push notifications.
+	 */
 	interface DeviceDetails {
+		/**
+		 * A unique identifier for the device generated by the device itself.
+		 */
 		id: string;
+		/**
+		 * Optional trusted [client identifier](https://ably.com/documentation/core-features/authentication#identified-clients) for the device.
+		 */
 		clientId?: string;
+		/**
+		 * Platform of the push device.
+		 */
 		platform: "android" | "ios" | "browser";
+		/**
+		 * Form factor of the push device.
+		 */
 		formFactor: "phone" | "tablet" | "desktop" | "tv" | "watch" | "car" | "embedded" | "other";
+		/**
+		 * Optional metadata object for this device. The metadata for a device may only be set by clients with `push-admin` privileges.
+		 */
 		metadata?: any;
+		/**
+		 * Secret value for the device.
+		 */
 		deviceSecret?: string;
 		push: DevicePushDetails;
 	}
 
+	/**
+	 * An object encapsulating the subscription of a device or group of devices sharing a client identifier to a channel in order to receive push notifications.
+	 */
 	interface PushChannelSubscription {
+		/**
+		 * The channel that this push notification subscription is associated with.
+		 */
 		channel: string;
+		/**
+		 * The device with this identifier is linked to this channel subscription. When present, `clientId` is never present.
+		 */
 		deviceId?: string;
+		/**
+		 * Devices with this client identifier are included in this channel subscription. When present, `deviceId` is never present.
+		 */
 		clientId?: string;
 	}
 
 	type DevicePushState = "ACTIVE" | "FAILING" | "FAILED";
 
 	interface DevicePushDetails {
+		/**
+		 * Push recipient details for this device. See the [REST API push publish documentation](https://ably.com/documentation/rest-api#message-extras-push) for more details.
+		 */
 		recipient: any;
+		/**
+		 * The current state of the push device.
+		 */
 		state?: DevicePushState;
+		/**
+		 * When the device’s state is failing or failed, this attribute contains the reason for the most recent failure.
+		 */
 		error?: ErrorInfo;
 	}
 
 	interface DeviceRegistrationParams {
+		/**
+		 * Filter to restrict to devices associated with the given client identifier
+		 */
 		clientId?: string;
+		/**
+		 * Filter to restrict to devices associated with the given device identifier.
+		 */
 		deviceId?: string;
+		/**
+		 * Maximum number of devices per page to retrieve, up to 1,000. Defaults to 100.
+		 */
 		limit?: number;
 		state?: DevicePushState;
 	}
 
 	interface PushChannelSubscriptionParams {
+		/**
+		 * Filter to restrict to subscriptions associated with the given channel.
+		 */
 		channel?: string;
+		/**
+		 * Filter to restrict to devices associated with the given client identifier. Cannot be used with a clientId param.
```

Comment:
```suggestion
		 * Filter to restrict to devices associated with the given client identifier. Cannot be used with a `clientId` param.
```

Though, actually, I'm not sure what this means... our current context is params and the property being documented is called `clientId`. Recursive reference, surely not?